diff --git a/scripts/build/toolchain/termux_setup_toolchain_29.sh b/scripts/build/toolchain/termux_setup_toolchain_29.sh
index 6abf7d5a63..edb273d4e5 100644
--- a/scripts/build/toolchain/termux_setup_toolchain_29.sh
+++ b/scripts/build/toolchain/termux_setup_toolchain_29.sh
@@ -128,18 +128,16 @@ termux_setup_toolchain_29() {
 		return
 	fi
 
-	[ -d "$TERMUX_STANDALONE_TOOLCHAIN" ] || mkdir -p "$TERMUX_STANDALONE_TOOLCHAIN"
-	[ -d "${TERMUX_STANDALONE_TOOLCHAIN}-upper" ] || mkdir -p "${TERMUX_STANDALONE_TOOLCHAIN}-upper"
-	[ -d "${TERMUX_STANDALONE_TOOLCHAIN}-work" ] || mkdir -p "${TERMUX_STANDALONE_TOOLCHAIN}-work"
-
-
-	if ! mountpoint -q "${TERMUX_STANDALONE_TOOLCHAIN}"; then
-		fuse-overlayfs \
-			"${TERMUX_STANDALONE_TOOLCHAIN}" \
-			-o lowerdir="${NDK}/toolchains/llvm/prebuilt/linux-x86_64" \
-			-o upperdir="${TERMUX_STANDALONE_TOOLCHAIN}-upper" \
-			-o workdir="${TERMUX_STANDALONE_TOOLCHAIN}-work"
-	fi
+	if [ ! -d "${TERMUX_STANDALONE_TOOLCHAIN}/bin" ]; then
+		rm -rf \
+			"$TERMUX_STANDALONE_TOOLCHAIN" \
+			"${TERMUX_STANDALONE_TOOLCHAIN}-upper" \
+			"${TERMUX_STANDALONE_TOOLCHAIN}-work"
+		mkdir -p "$TERMUX_STANDALONE_TOOLCHAIN"
+		cp -a \
+			"${NDK}/toolchains/llvm/prebuilt/linux-x86_64"/. \
+			"${TERMUX_STANDALONE_TOOLCHAIN}/"
+	fi
 
 	if [ -f "${TERMUX_STANDALONE_TOOLCHAIN}/.termux-standalone-toolchain" ]; then
 		return
