diff --git a/packages/sed/build.sh b/packages/sed/build.sh
index 2ad4862dd0..dd975ab5e4 100644
--- a/packages/sed/build.sh
+++ b/packages/sed/build.sh
@@ -9,6 +9,7 @@ TERMUX_PKG_SHA256=6e226b732e1cd739464ad6862bd1a1aba42d7982922da7a53519631d24975
 TERMUX_PKG_ESSENTIAL=true
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_GROUPS="base-devel"
+TERMUX_PKG_EXTRA_MAKE_ARGS="-f Makefile"
 
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
 ac_cv_func_nl_langinfo=no
@@ -20,5 +21,6 @@ termux_step_pre_configure() {
 }
 
 termux_step_post_configure() {
+	rm -f "$TERMUX_PKG_SRCDIR/GNUmakefile"
 	touch -d "next hour" $TERMUX_PKG_SRCDIR/doc/sed.1
 }
