--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -1143,6 +1143,23 @@
           'conditions': [
             ['v8_enable_webassembly==1', {
               'conditions': [
+                ['_toolset=="host" and OS=="android" and host_arch=="arm64" and (host_os=="darwin" or host_os=="linux")', {
+                  'sources': [
+                    '<(V8_ROOT)/src/trap-handler/handler-inside-posix.cc',
+                    '<(V8_ROOT)/src/trap-handler/handler-outside-posix.cc',
+                  ],
+                }],
+                ['_toolset=="host" and OS=="android" and host_arch=="x64" and (host_os=="darwin" or host_os=="linux")', {
+                  'sources': [
+                    '<(V8_ROOT)/src/trap-handler/handler-inside-posix.cc',
+                    '<(V8_ROOT)/src/trap-handler/handler-outside-posix.cc',
+                  ],
+                }],
+                ['_toolset=="host" and OS=="android" and host_arch=="x64" and (host_os=="darwin" or host_os=="linux")', {
+                  'sources': [
+                    '<(V8_ROOT)/src/trap-handler/handler-outside-simulator.cc',
+                  ],
+                }],
                 ['((_toolset=="host" and host_arch=="arm64" or _toolset=="target" and target_arch=="arm64") and (OS=="linux" or OS=="mac" or OS=="ios")) or ((_toolset=="host" and host_arch=="x64" or _toolset=="target" and target_arch=="x64") and (OS=="linux" or OS=="mac"))', {
                   'sources': [
                     '<(V8_ROOT)/src/trap-handler/handler-inside-posix.cc',
@@ -1445,15 +1445,22 @@
             ],
           },
           'target_conditions': [
-            ['_toolset=="host"', {
+            ['_toolset=="host" and host_os=="darwin"', {
+              'sources': [
+                '<(V8_ROOT)/src/base/debug/stack_trace_posix.cc',
+                '<(V8_ROOT)/src/base/platform/platform-darwin.cc',
+              ],
+            }],
+            ['_toolset=="host" and host_os=="linux"', {
               'sources': [
                 '<(V8_ROOT)/src/base/debug/stack_trace_posix.cc',
                 '<(V8_ROOT)/src/base/platform/platform-linux.cc',
               ],
-            }, {
+            }],
+            ['_toolset=="target"', {
               'sources': [
                 '<(V8_ROOT)/src/base/debug/stack_trace_android.cc',
                 '<(V8_ROOT)/src/base/platform/platform-linux.cc',
               ],
             }],
           ],
